Scripts/RemediationScripts/Remediate-AddDiagnosticSettingsForLogicApps.ps1

```diff
@@ -0,0 +1,507 @@
+
+    Write-Host "*** To add diagnostic seeting for Logic Apps in a Subscription, Contributor and higher privileges on the Resource Groups containing Logic Apps in the Subscription is required. ***" -ForegroundColor $([Constants]::MessageType.Info)
+    
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 2 of 4] Fetching Logic Apps..."
+
+    $logicAppResources = @()
+
+    # No file path provided as input to the script. Fetch all Logic Apps in the Subscription.
+    if ([String]::IsNullOrWhiteSpace($FilePath)) {
+        Write-Host "Fetching all Logic Apps in Subscription: $($context.Subscription.SubscriptionId)" -ForegroundColor $([Constants]::MessageType.Info)
+
+        # Get all Logic Apps in a Subscription.
+        $logicApps = Get-AzResource -ResourceType "Microsoft.Logic/workflows" -ErrorAction Stop 
+        
+        $logicAppResources += $logicApps | Select-Object @{N = 'ResourceId'; E = { $_.ResourceId } },
+        @{N = 'ResourceGroupName'; E = { $_.ResourceGroupName } },
+        @{N = 'Name'; E = { $_.ResourceName } },
+        @{N = 'ResourceType'; E = { $_.ResourceType } },
+        @{N = 'Location'; E = { $_.Location } }
+    }
+    else {
+        if (-not (Test-Path -Path $FilePath)) {
+            Write-Host "ERROR: Input file - $($FilePath) not found. Exiting..." -ForegroundColor $([Constants]::MessageType.Error)
+            break
+        }
+
+        Write-Host "Fetching all Logic Apps from $($FilePath)" -ForegroundColor $([Constants]::MessageType.Info)
+
+        # Importing the list of Logic Apps to be remediated.
+        $logicAppDetails = Import-Csv -LiteralPath $FilePath
+
+        $logicAppResources = $logicAppDetails | Where-Object { ![String]::IsNullOrWhiteSpace($_.ResourceId) }
+    }
+
+    $totalLogicApps = $logicAppResources.Count
+
+    if ($totalLogicApps -eq 0) {
+        Write-Host "No Logic Apps found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        break
+    }
+
+    Write-Host "Found $($totalLogicApps) Logic Apps." -ForegroundColor $([Constants]::MessageType.Update)
+    Write-Host "Fetching Logic App diagnostic settings without required diagnostic setting..." -ForegroundColor $([Constants]::MessageType.Update)
+    # Includes Logic Apps where required diagnostic settings are present.
+    $logicAppsWithRequiredDiagnosticSetting = @()
+
+    # Includes Logic Apps where required diagnostic settings are not present.
+    $logicAppsWithoutRequiredDiagnosticSetting = @()
+
+    # Check if required Diagnostic setting for Logic App is present
+
+    $logicAppResources | ForEach-Object {
+        try {
+
+            $settings = Get-AzDiagnosticSetting -ResourceId $_.ResourceId
+            
+            if ($null -eq $settings) {
+                # No diagnostic setting is configured
+                $logicAppsWithoutRequiredDiagnosticSetting += $_
+            }
+            else {
+                # Diagnostic settings includes WorkflowRuntime logs
```

Comment:
IF workflowruntime is true or allLogs is true it represent complaint state ryt? So we can have single if with or condition.

Reply:
updated thanks